benchmark/src/main/java/ai/timefold/solver/benchmark/impl/aggregator/BenchmarkAggregator.java

```diff
@@ -40,11 +48,145 @@ public void setBenchmarkReportConfig(BenchmarkReportConfig benchmarkReportConfig
+
+            if (selectedDirectoryNames.contains(benchmarkDirectoryName)) {
+                LOGGER.info("Including benchmark results from directory: {}", benchmarkDirectoryName);
+
+                for (SolverBenchmarkResult solverResult : plannerResult.getSolverBenchmarkResultList()) {
+                    selectedSingleBenchmarkResults.addAll(solverResult.getSingleBenchmarkResultList());
+                    solverBenchmarkResultNameMap.put(solverResult, solverResult.getName());
+                }
+            }
+        }
+
+        if (selectedSingleBenchmarkResults.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "No valid benchmark results found for selected directories: " + selectedDirectoryNames);
+        }
+
+        // Configure the aggregator instance
+        this.setBenchmarkDirectory(benchmarkDirectory);
+        this.setBenchmarkReportConfig(benchmarkConfig.getBenchmarkReportConfig());
+
+        // Perform the aggregation with selected results - returns HTML report file
+        return this.aggregate(selectedSingleBenchmarkResults, solverBenchmarkResultNameMap);
+    }
+
+    public List<String> getAvailableBenchmarkDirectories(PlannerBenchmarkConfig benchmarkConfig) {
+        File benchmarkDirectory = benchmarkConfig.getBenchmarkDirectory();
+        List<String> directories = new ArrayList<>();
+
+        if (!benchmarkDirectory.exists() || !benchmarkDirectory.isDirectory()) {
+            return directories;
+        }
+
+        File[] subdirs = benchmarkDirectory.listFiles(File::isDirectory);
+        if (subdirs != null) {
+            for (File subdir : subdirs) {
+                // Only include directories that have a benchmark result file
+                File resultFile = new File(subdir, "plannerBenchmarkResult.xml");
+                if (resultFile.exists()) {
+                    directories.add(subdir.getName());
+                }
+            }
+        }
+
+        return directories;
+    }
+
+    public File aggregateAndShowReportInBrowser(PlannerBenchmarkConfig benchmarkConfig) {
+        File benchmarkDirectoryPath = this.aggregateBenchmarks(benchmarkConfig);
+        this.showReportInBrowser();
+        return benchmarkDirectoryPath;
+    }
+
+    private void showReportInBrowser() {
+        if (this.benchmarkReport == null) {
+            throw new IllegalStateException("No benchmark report available. Run aggregation first.");
+        }
+
+        File htmlOverviewFile = this.benchmarkReport.getHtmlOverviewFile();
+        Desktop desktop = Desktop.isDesktopSupported() ? Desktop.getDesktop() : null;
+        if (desktop != null && desktop.isSupported(Action.BROWSE)) {
+            try {
+                desktop.browse(htmlOverviewFile.getAbsoluteFile().toURI());
+            } catch (IOException e) {
+                throw new IllegalStateException(
+                        "Failed showing htmlOverviewFile (" + htmlOverviewFile + ") in the default browser.", e);
```

Comment:
The solver recently started using a different pattern for exceptions, which improves their readability if there are more arguments.

    throw new IllegalStateException(
                        "Failed showing htmlOverviewFile (%s) in the default browser.".formatted(e)) 

Please use the new pattern and apply it consistently.